wgpu-core/src/command/memory_init.rs

```diff
@@ -3,23 +3,22 @@ use std::{collections::hash_map::Entry, ops::Range, vec::Drain};
 use hal::CommandEncoder;
 
 use crate::{
-    command::collect_zero_buffer_copies_for_clear_texture,
     device::Device,
     hub::Storage,
     id::{self, TextureId},
     init_tracker::*,
     resource::{Buffer, Texture},
-    track::{ResourceTracker, TextureSelector, TextureState, TrackerSet},
-    FastHashMap,
+    track::{ResourceTracker, TextureState, TrackerSet},
+    FastHashMap, Stored,
 };
 
-use super::{BakedCommands, DestroyedBufferError, DestroyedTextureError};
+use super::{clear::clear_texture, BakedCommands, DestroyedBufferError, DestroyedTextureError};
 
 /// Surface that was discarded by `StoreOp::Discard` of a preceding renderpass.
 /// Any read access to this surface needs to be preceded by a texture initialization.
 #[derive(Clone)]
 pub(crate) struct TextureSurfaceDiscard {
-    pub texture: TextureId,
+    pub texture: Stored<TextureId>,
```

Comment:
where is this refcount used?

Reply:
while trying to elaborate why this is needed for the barrier in clear_texture_no_device I was able to remove it again :)
A tad bit messy though, see comment here https://github.com/Wumpf/wgpu/commit/5073b93b45d8e68b0e6f61a4e257a53591cb6257#diff-0935c4873f626097404cc18bab07ee4833b98c9fb868f1f30cce86d053b005bbR285-R290